internal/versionsapi/cli/list.go

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/edgelesssys/constellation/v2/internal/logger"
+	"github.com/edgelesssys/constellation/v2/internal/versionsapi"
+	verclient "github.com/edgelesssys/constellation/v2/internal/versionsapi/client"
+	"github.com/rogpeppe/go-internal/semver"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap/zapcore"
+)
+
+func newListCmd() *cobra.Command {
```

Comment:
Shouldn't `versionsapi list` return all released image versions? Or do we only support starting from v2.3.0?

Reply:
When specifying a different minor version I get an error:
```
./versionsapi list --minor-version v2.2
Error: listing patch versions: the requested resource was not found: operation error S3: GetObject, https response error StatusCode: 404, RequestID: 7VMY5X0H9XV3MR0K, HostID: Zi5AIOzEu4bAnRx73CtvaMEABYzW1sLmy5p3e+Pr+sSL5bjyRBlhaRWfZlZWk0PWjoSjs9PnoI3JKUJz44Y5aw==, NoSuchKey: 
```

Should we add a check to let the user know this tool does not support old images?

Reply:
> Shouldn't versionsapi list return all released image versions?

No, only the ones versioned under this version of the versions API. Note that there will be another breaking change following this PR, see discussion in #788 .

Reply:
Maybe list shouldn't error in case no version table is found and instead return nothing (as the list of versions available is empty).

Reply:
Done in [db4e826](https://github.com/edgelesssys/constellation/pull/840/commits/db4e8265a806f5839eac122e16e46de6b91aff6d)